chore: auth-wraper docs

pedroharbs opened the pull request:
- Added auth-wrapper docs.

pedroharbs committed:
chore: adding auth-wrapper docs

pedroharbs committed:
chore: updated README.md with authentication

sowmyaraj92 committed:
review edits - 1

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+```
+​
+You can find a full sample application [here]()
+​
````
Missing link

cognite-ornellas replied:
Is this resolved? Could not find it.

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+​
+```
+​
+You can find a full sample application [here]()
````
Missing link

cognite-ornellas replied:
Is this resolved? Could not find it.

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+```
+​
+You can find a full sample application [here]()
+​
````
Missing link

cognite-ornellas replied:
Is this resolved? Could not find it.

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+```
+​
+You can find a full sample application [here]()
+​
````
Missing link

cognite-ornellas replied:
Is this resolved? Could not find it.

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+​
+```
+​
+You can find a full sample application [here]()
````
Missing link

cognite-ornellas replied:
Is this resolved? Could not find it.

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+```
+​
+You can find a full sample application [here]()
+​
````
Missing link

sowmyaraj92 commented on guides/auth-wrapper.md:
````diff
@@ -0,0 +1,341 @@
+```
+​
+You can find a full sample application [here]()
+​
````
Missing link

sowmyaraj92 commented on guides/auth-wrapper.md:
```diff
@@ -0,0 +1,341 @@
+### getToken metod
+​
+API keys use the same API in the SDK, using `getToken` you will need to set the additional flag `apiKeyMode` to
+true to handle headers appropriately.
```
Unclear information - need more context here.

sowmyaraj92 commented on guides/auth-wrapper.md:
```diff
@@ -0,0 +1,341 @@
+​
+## API keys
+​
+### getToken metod
```
metod > "method"

sowmyaraj92 commented on guides/auth-wrapper.md:
```diff
@@ -0,0 +1,341 @@
+​
+### credentials metod
+​
+API keys use the same API in the SDK, using `credentials` you will need to set the following object to handle headers appropriately.
```
Same as the above comment.

sowmyaraj92 commented on guides/auth-wrapper.md:
```diff
@@ -0,0 +1,337 @@
+  - [getToken metod](#gettoken-metod)
+  - [credentials method](#credentials-method)
+- [Manually trigger authentication](#manually-trigger-authentication)
+​
```
Todo:- Will update the entire header in the next review commit!

sowmyaraj92 committed:
chore: review edits 2

pedroharbs committed:
feat: auth wrapper as a provider (#816)

* chore: adding fields to verify in each OIDC method

* feat: removed auth-wrapper from inside SDK and added as a provider

* test: updating tests

* style: added eslint fix to baseCogniteClient

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-beta@5.1.1
 - @cognite/sdk-core@4.4.0
 - @cognite/sdk-playground@5.3.1
 - @cognite/sdk@7.6.1

Commit:
feat(annotations): remove linked resource fields (#822)

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-playground@5.4.0

KissSpinobleZsolt committed:
feat: noAuthMode support (#824)

* added oauth support

* make it not required

* fix(nauth): noauth mode

* feat: added noauth support

* fix: make option not required

* fix(nauth): noauth mode

* fix(format): revert

* feat: noAuthMode execution and adding unit tests

* fix: update baseCogniteClient.ts format

Co-authored-by: Murad Sæter <murad.sater@cognite.com>
Co-authored-by: cognite-ornellas <rodrigo.ornellas@cognite.com>
Co-authored-by: Murad Sæter <51971968+muradsater@users.noreply.github.com>

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-beta@5.1.2
 - @cognite/sdk-core@4.5.0
 - @cognite/sdk-playground@5.4.1
 - @cognite/sdk@7.6.2

emmabuoen committed:
feat(annotations): move annotations to stable (#823)

* refactor: move files and add annotationsapi to cognite client

* refactor: fix imports

* refactor: remove annotations from playground

* refactor: add annotation types to main types

* refactor: cleanup imports

* refactor: lint

* refactor: remove types file

cognite-cicd committed:
chore(release): publish new package versions [skip ci]

 - @cognite/sdk-beta@5.1.3
 - @cognite/sdk-playground@5.5.0
 - @cognite/sdk@7.7.0

cognite-ornellas committed:
feat: add anders as codeowner to codegen (#825)

* feat:  add anders as codeowner to codegen

* feat: adding anders to playground and stable

Co-authored-by: cognite-ornellas <rodrigo.ornellas@cognite.com>

andersfylling committed:
chore: update CODEOWNERS (#826)

pedroharbs committed:
chore: updating auth-wrapper docs

pedroharbs committed:
chore: updating auth-wrapper docs

pedroharbs committed:
Merge branch 'master' into feat/refresh_token_docs

sowmyaraj92 committed:
Merge branch 'feat/refresh_token_docs' of https://github.com/cognitedata/cognite-sdk-js into feat/refresh_token_docs

sowmyaraj92 committed:
review edits - 3

sowmyaraj92 committed:
chore: review edits

sowmyaraj92 committed:
getting started section

sowmyaraj92 committed:
chore: getting started section

sowmyaraj92 committed:
chore: install dependencies

sowmyaraj92 committed:
chore: install dependencies

pedroharbs committed:
chore: adding scripts info to auth-wrapper guide

pedroharbs committed:
chore: adding sample to client_credentials

pedroharbs committed:
Merge branch 'master' into feat/refresh_token_docs

sowmyaraj92 committed:
Merge branch 'feat/refresh_token_docs' of https://github.com/cognitedata/cognite-sdk-js into feat/refresh_token_docs

sowmyaraj92 committed:
chore: updating intro

sowmyaraj92 reviewed (changes requested):
@cognite-ornellas There are a few places where we need more context and clarity in the PR, let me know when you're done making the changes.

sowmyaraj92 committed:
Merge branch 'master' into feat/refresh_token_docs

Commit:
chore: adjusting documentation

sowmyaraj92 committed:
chore: updating content + final edits

sowmyaraj92 committed:
Merge branch 'master' into feat/refresh_token_docs

sowmyaraj92 committed:
chore: fixing articles

sowmyaraj92 committed:
Merge branch 'feat/refresh_token_docs' of https://github.com/cognitedata/cognite-sdk-js into feat/refresh_token_docs

sowmyaraj92 commented on README.md:
```diff
@@ -24,6 +24,10 @@ There are small bare-bones javascript (and typescript) projects in the `samples/
+## Authentication
+
+We offer two ways to get logged on our sdk, the first and most recommended is using [@cognite/auth-wrapper](./guides/auth-wrapper.md)
+and the second is using [Microsoft MSAL Library](./guides/auth-wrapper.md)
```
```suggestion
We offer two ways to sign in to our SDK. The first and recommended approach is using [@cognite/auth-wrapper](./guides/auth-wrapper.md)
and the second is using [Microsoft MSAL Library](./guides/auth-wrapper.md)
```

sowmyaraj92 reviewed (approved):
Just one minor change, otherwise we are good to merge! LGTM 🚀

sowmyaraj92 committed:
Update README.md
